src/domain/core/src/services/compact_service.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::path::Path;
+use std::sync::Arc;
+
+use opendatafabric::*;
+use thiserror::Error;
+
+use crate::*;
+
+#[async_trait::async_trait]
+pub trait CompactService: Send + Sync {
+    async fn compact_dataset(
+        &self,
+        dataset_handle: &DatasetHandle,
+        dataset_dir_path: &Path,
```

Comment:
Passing a path might be fine for local FS repository. What are you going to do with S3?
The repository actualy knows where it is located on the disk very well, I don't think you need to pass this info here.

Reply:
Changed to run dir